fn main() {
    a.5.2E+
    //~^ ERROR: unexpected token: `5.2E+`
    //~| ERROR: expected one of `.`, `;`, `?`, `}`, or an operator, found `5.2E+`
    //~| ERROR: expected at least one digit in exponent
}

// ferrocene-annotations: fls_18k3uajrgq5f
// Field Access Expressions
//
// ferrocene-annotations: fls_6ydylimiv553
// Place Expressions
//
// ferrocene-annotations: fls_xcwfotmq2e5d
// Field Resolution
