fn main() { let x = 1.0_f32 || 2.0_f32; }
//~^ ERROR mismatched types
//~| ERROR mismatched types

// ferrocene-annotations: fls_lstusiu2c8lu
// Lazy Boolean Expressions
//
// ferrocene-annotations: fls_izdv9i4spokw
// Operator Expressions
